expat/CMakeLists.txt

```diff
@@ -395,6 +395,12 @@ if(EXPAT_SHARED_LIBS)
     if(MSVC)
         set(expat_SRCS ${expat_SRCS} lib/libexpat.def)
     endif()
+    if(WIN32)
+        # Add DLL version
+        string(REPLACE "." "," _EXPAT_DLL_VERSION ${PROJECT_VERSION}.0)
+        add_definitions(-DVER_FILEVERSION=${_EXPAT_DLL_VERSION})
```

Comment:
@petitlapin thabnks for the adjustments!

Did you see my comment on `add_definitions` being global? Would you mind using `target_compile_definitions` for target `expat` instead, e.g.
  `target_compile_definitions(expat PRIVATE VER_FILEVERSION=${_EXPAT_DLL_VERSION})`?
The `-D` is not needed in that variation.
https://cmake.org/cmake/help/latest/command/target_compile_definitions.html

Reply:
Yes, I wrote directly above as it is the only point that does not work directly (I'm resolving the other point as we'll continue the discussion here): 

The second point (target_compile_definitions) does not seem to work. When doing so, the compiler does not replace the defines: C:\libexpat\expat\win32\version.rc(2): error RC2104: undefined keyword or key name: VER_FILEVERSION [C:\libexpat\build_msvc\expat.vcxproj]

Another solution used in the [CMake code directly](https://gitlab.kitware.com/cmake/cmake/-/blob/master/Source/CMakeLists.txt#L1201) is to pass via a configure_file to avoid the add_definitions.

I can take a look (tomorrow or this week-end) if it is preferred.

Reply:
@petitlapin I see :thinking: Let me think more about this, I'm not sure yet. I would love to understand why `target_compile_definitions` doesn't work as expected. Maybe some detail is off.

Reply:
My bad, I made a typo on the variable name...

I had to add it for the tests also.